auth/token.go

```diff
@@ -41,6 +34,15 @@ type Token interface {
 	GetClaimAsString(claim string) (string, error)        // GetClaimAsString returns a custom claim type asserted as string. Returns error if the claim is not available or not a string.
 	GetClaimAsStringSlice(claim string) ([]string, error) // GetClaimAsStringSlice returns a custom claim type asserted as string slice. The claim name is case sensitive. Returns error if the claim is not available or not an array
 	GetAllClaimsAsMap() map[string]interface{}            // GetAllClaimsAsMap returns a map of all claims contained in the token. The claim name is case sensitive. Includes also custom claims
+	// only for internal usage
+	audience() []string    // audience returns "aud" claim, if it doesn't exist empty string is returned
+	expiration() time.Time // expiration returns "exp" claim, if it doesn't exist empty string is returned
+	isExpired() bool       // isExpired returns true, if 'exp' claim + leeway time of 1 minute is before current time
```

Comment:
this one should be exposed

Reply:
Why? If all the validation should be done within the (blackbox) client lib it does not make sense to let the application validate single parts of it.

Reply:
I had many requests in the past, that the clientlib wanted access to token internals.
So I'd also say... just provide getter... this is just a simple method and no "validation" step. Otherwise these customers will parse the token on their own...

Reply:
the `getExpired` is needed, in case customers have their own token caches.